sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -246,8 +246,6 @@ abstract class Optimizer(catalogManager: CatalogManager)
       CollapseProject,
       RemoveRedundantAliases,
       RemoveNoopOperators) :+
-    Batch("InsertMapSortInGroupingExpressions", Once,
```

Comment:
is it possible that new Aggregates are created after `FinishAnalysis`?

Reply:
It's good point. It may introduce new Aggregates in optimization, e.g., rewrite distinct, but seems not affect map type.  We disallow these operators on map type:
`Cannot have MAP type columns in DataFrame which calls set operations (INTERSECT, EXCEPT, etc.),`

Reply:
ok maybe, but is it safer to run `PullOutGroupingExpressions` again here? Or simply update `InsertMapSortInGroupingExpressions` to create a project to add `MapSort`.

Reply:
It requires more rules to run `PullOutGroupingExpressions` again, e.g., we need to re prune and ollapse project since we generate a new pre project. I changed to use such batch and add some tests for it.
```
Batch("AddMapSortInAggregate", FixedPoint(1),
  AddMapSortInAggregate,
  PullOutGroupingExpressions,
  ColumnPruning,
  CollapseProject)
```